cadquery/selectors.py

```diff
@@ -155,287 +154,307 @@ def isInsideBox(p):
 
 class BaseDirSelector(Selector):
     """
-        A selector that handles selection on the basis of a single
-        direction vector
+    A selector that handles selection on the basis of a single direction vector.
     """
 
-    def __init__(self, vector, tolerance=0.0001):
+    def __init__(self, vector: Vector, tolerance: float = 0.0001):
         self.direction = vector
-        self.TOLERANCE = tolerance
+        self.tolerance = tolerance
 
-    def test(self, vec):
+    def test(self, vec: Vector) -> bool:
         "Test a specified vector. Subclasses override to provide other implementations"
         return True
 
-    def filter(self, objectList):
+    def filter(self, objectList: Sequence[Shape]) -> List[Shape]:
         """
-            There are lots of kinds of filters, but
-            for planes they are always based on the normal of the plane,
-            and for edges on the tangent vector along the edge
+        There are lots of kinds of filters, but for planes they are always
+        based on the normal of the plane, and for edges on the tangent vector
+        along the edge
         """
         r = []
         for o in objectList:
             # no really good way to avoid a switch here, edges and faces are simply different!
-
-            if type(o) == Face:
-                # a face is only parallell to a direction if it is a plane, and its normal is parallel to the dir
-                normal = o.normalAt(None)
-
-                if self.test(normal):
-                    r.append(o)
-            elif type(o) == Edge and (
+            if isinstance(o, Face):
+                # a face is only parallel to a direction if it is a plane, and its normal is parallel to the dir
+                test_vector = o.normalAt(None)
+            elif isinstance(o, Edge) and (
                 o.geomType() == "LINE" or o.geomType() == "PLANE"
```

Comment:
@adam-urbanczyk, I see you wrote this line 4 years ago. I don't think an Edge can have `geomType() == "PLANE"`. Was this a typo or am I wrong?

Reply:
Looks indeed like a mistake.

Reply:
OK, I'll fix it and make a test for it. I'll write a more detailed comment when I'm done.